packages/vanilla-dom-utils/src/events.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import { hashString } from "@vanilla/utils";
+
+const delegatedEventListeners = {};
+
+/**
+ * Create an event listener using event delegation.
+ *
+ * @param eventName - The Event to listen for.
+ * @param filterSelector - A CSS selector to match against.
+ * @param callback - The callback function. This gets passed the fired event.
+ * @param scopeSelector - And element to scope the event listener to.
+ *
+ * @returns The hash of the event. Save this to use removeDelegatedEvent().
+ */
+export function delegateEvent(
+    eventName: string,
+    filterSelector: string,
+    callback: (event: Event, triggeringElement: HTMLElement) => boolean | void,
+    scopeSelector?: string | HTMLElement,
+): string | undefined {
```

Comment:
In which cases this function can return `undefined`?

Reply:
I cannot. Good catch.

Reply:
I think that it could at some point in the past but was changed a few months ago.